File: backend/Services/AuthService.cs
using EmployeeHub.Api.DTOs;
using EmployeeHub.Api.Models;
using EmployeeHub.Api.Repositories;
using Microsoft.AspNetCore.Identity;

/*
 * AuthService.cs contains the authentication business logic.
 *
 * Responsibility:
 * - Register new users.
 * - Verify login credentials.
 * - Ask JwtTokenService to create a token after successful authentication.
 *
 * Connection to other files:
 * - AuthController receives HTTP requests and calls this service.
 * - UserRepository communicates with SQL Server.
 * - JwtTokenService creates the JWT returned to React.
 */
namespace EmployeeHub.Api.Services;

public class AuthService(
    IUserRepository userRepository,
    PasswordHasher<User> passwordHasher,
    IJwtTokenService jwtTokenService) : IAuthService
{
    public async Task<AuthResponse> RegisterAsync(RegisterRequest request)
    {
        if (await userRepository.EmailExistsAsync(request.Email))
        {
            throw new InvalidOperationException("A user with this email already exists.");
        }

        var user = new User
        {
            FullName = request.FullName.Trim(),
            Email = request.Email.Trim().ToLower()
        };

        user.PasswordHash = passwordHasher.HashPassword(user, request.Password);

        var savedUser = await userRepository.AddAsync(user);
        return CreateAuthResponse(savedUser);
    }

    public async Task<AuthResponse?> LoginAsync(LoginRequest request)
    {
        var user = await userRepository.GetByEmailAsync(request.Email);

        if (user is null)
        {
            return null;
        }

        var passwordResult = passwordHasher.VerifyHashedPassword(user, user.PasswordHash, request.Password);

        if (passwordResult == PasswordVerificationResult.Failed)
        {
            return null;
        }

        return CreateAuthResponse(user);
    }

    private AuthResponse CreateAuthResponse(User user)
    {
        return new AuthResponse(
            jwtTokenService.CreateToken(user),
            user.Email,
            user.FullName);
    }
}

File: frontend/src/services/authService.ts
import { apiClient } from '../api/apiClient'
import type { AuthResponse, LoginRequest, RegisterRequest } from '../types/auth'

/*
 * authService.ts contains frontend functions for authentication API calls.
 *
 * Responsibility:
 * - Send login and register requests with Axios.
 * - Return the JWT response to React hooks/components.
 *
 * Connection to other files:
 * - AuthPage.tsx calls these functions through useAuth.ts.
 * - AuthController.cs receives these requests in ASP.NET Core.
 */
export async function login(request: LoginRequest): Promise<AuthResponse> {
  const response = await apiClient.post<AuthResponse>('/api/auth/login', request)
  return response.data
}

export async function register(request: RegisterRequest): Promise<AuthResponse> {
  const response = await apiClient.post<AuthResponse>('/api/auth/register', request)
  return response.data
}
